### PR TITLE
Baseline

### LINKED ISSUE #7: Backpropogation issues
Implemented algorithm does not working while testing mnist dataset(

### PR DESCRIPTION
**Pull request by kuweg**

### CODE REVIEW

**Comment by kuweg** [comment]
Neural network succesfull learn MNIST with 91% accuracy. Confusion matrix looks good too - 1 and 2 types error are small
image

**Comment by kuweg** [comment]
TODO:
Optimize model - a single sample per gradient step takes too much time (approx 5-10 seconds per epoch for 31500 samples)
Model structure is not optimal for this type of problem - MSE and tanh -> could be CategoricalCrossEntropy and softmax
Make Sequential model class more flexible in term of usage -> loss and optimizer could be added after model initialization, but it's optinal
Rework MNIST loader util - it does not look conviniet to localy store data -> add loader and unzip

**Comment by kuweg** [comment]
purenn 1.0????
